src/v/kafka/requests/fetch_request.cc

```diff
@@ -428,92 +428,112 @@ handle_ntp_fetch(op_context& octx, model::ntp ntp, fetch_config config) {
           try {
               auto response = f.get0();
               response.id = p_id;
-              octx.add_partition_response(std::move(response));
+              octx.set_partition_response(std::move(response));
           } catch (...) {
               /*
                * TODO: this is where we will want to handle any storage
                * specific errors and translate them into kafka response
                * error codes.
                */
-              octx.response_error = true;
-              octx.add_partition_response(make_partition_response_error(
+              octx.set_partition_response(make_partition_response_error(
                 error_code::unknown_server_error));
           }
       });
 }
 
+static ss::future<> fetch_topic_partition(
```

Comment:
this seems critical. we should try to push a version next tuesday?

Reply:
Yes. It is critical